src/bandersnatch/simple.py

```diff
@@ -114,7 +111,11 @@ def get_simple_dirs(self, simple_dir: Path) -> List[Path]:
             # We are using index page directory hashing, so the directory
             # format is /simple/f/foo/.  We want to return a list of dirs
             # like "simple/f".
-            subdirs = [simple_dir / x for x in simple_dir.iterdir() if x.is_dir()]
+            subdirs = sorted(
+                simple_dir / str(x.name)
```

Comment:
Isn't Path().name already a str? Maybe not for some of the storage plugins or protocol you just added? Why you casting to a str everywhere?

Reply:
Because `os.DirEntry.name` have the type `AnyStr = TypeVar('AnyStr', bytes, str)`, we have to `str()` here.